### DISCUSSION TITLE
how do I achieve graphics that work on real hardware?

### DISCUSSION THREAD

**Discussion by jspa2** [CosmosOS/Cosmos#1616] [Q&A]
I only need 320x200.
I'm seeing a lot of functions to get graphics working after searching, but I'm not sure which of them still work / if they work on real hardware. Help appreciated :)

Edit: I'm using a devkit built yesterday

**Comment by quajak** [comment]
[`GetFullScreenCanvas`](https://cosmosos.github.io/api/Cosmos.System.Graphics.FullScreenCanvas.html#Cosmos_System_Graphics_FullScreenCanvas_GetFullScreenCanvas_Cosmos_System_Graphics_Mode_) with mode set to 320x200 should work. On real hardware currently on the VGA works, which means that under the hood it will just give you a [VGACanvas](https://github.com/CosmosOS/Cosmos/blob/master/source/Cosmos.System2/Graphics/VGACanvas.cs) . If you want to test your driver first in a vm its a good idea to directly instantiate the `VGACanvas` since `GetFullScreenCanvas` will use a different driver depending on the vm.

**Comment by jspa2** [comment]
Thanks. This is working very well
